File: android-shared-lib/src/main/java/com/skedgo/android/common/model/Availability.java
package com.skedgo.android.common.model;

public enum Availability {

  Available("AVAILABLE"),
  MissedPrebookingWindow("MISSED_PREBOOKING_WINDOW"),
  Cancelled("CANCELLED");

  private String value;

  Availability(String availability) {value = availability;}

  String getValue() {return value;}

  public static Availability fromString(String value) {
    for (Availability name : values()) {
      if (name.value.equals(value)) {
        return name;
      }
    }
    return null;
  }
}

File: android-shared-lib/src/main/java/com/skedgo/android/common/model/Availability.kt
package com.skedgo.android.common.model

enum class Availability(val value: String) {
  Available("AVAILABLE"),
  MissedPrebookingWindow("MISSED_PREBOOKING_WINDOW"),
  Cancelled("CANCELLED");

  companion object {
    @JvmStatic fun fromString(value: String?): Availability? {
      return values().firstOrNull { it.value == value }
    }
  }
}